src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -16,6 +20,7 @@ internal class RemoteJSRuntime : JSRuntime
         private readonly CircuitOptions _options;
         private readonly ILogger<RemoteJSRuntime> _logger;
         private CircuitClientProxy _clientProxy;
+        private ConcurrentDictionary<long, (Stream Stream, bool LeaveOpen)> _pendingStreams = new();
```

Comment:
Consider Pipes instead of Streams? You wouldn't need to manage buffers in SendDotNetStreamAsync

Reply:
The `Stream` instance here comes from user code, which is a more familiar and flexible API to offer.

Were you suggesting we should use `Pipe` as the API in user code (as in, they must supply a `Pipe` to us), or were you suggesting this for only if it was an internal detail?

Reply:
> were you suggesting this for only if it was an internal detail?

Right, I don't know the details of what users call, this just looked like a mostly internal thing.